### ISSUE TITLE
@typedef is only displayed as three dots

### ISSUE THREAD

**Issue by kungfooman** [TypeStrong/typedoc#2454]
## Search terms

@typedef, three points only, missing documentation for typedef tag

## Expected Behavior

I expect that all properties of the given `@typedef` are listed.

Alternatively a link to a new file explaining this typedef.

Or simply both - make it an option whether a typedef is inlined or creating a new typedef info file.

## Actual Behavior

A full `@typedef` definition is only producing three dots:

![image](https://github.com/TypeStrong/typedoc/assets/5236548/b16bed26-6400-47ca-bac9-96b5432f4cef)

## Steps to reproduce the bug

I use this code:

```js
/**
 * Animal counting object for a zoo.
 *
 * @typedef {object} AnimalCount
 * @property {number} elephant - The number of elephants.
 * @property {number} dolphins - The number of dolphins.
 */

class Location {
    /** @type {AnimalCount} */
    zoo = { elephant: 1, dolphins: 2 };
}
```

## Environment

-   Typedoc version: npm install typedoc@^0.25.4 (same issue on 0.25.1)
-   TypeScript version: ^5.2.2
-   Node.js version: v18.12.1
-   OS: Linux

Full config and repo: https://github.com/playcanvas/engine/blob/main/package.json

**Comment by Gerrit0** [comment]
typedefs will be converted as interfaces -- so long as they are exported.

If I copy your example into a `.js` file (and export `Location`), and run TypeDoc on it, I get the following, with a clickable link to `AnimalCount`

![image](https://github.com/TypeStrong/typedoc/assets/19329837/17c82ad7-2ab7-4181-8a5c-6a1d2c931284)

JS projects are really annoying to maintain with typedefs since they can't easily re-export types like TS projects can if the entry point doesn't export *everything* from the sub-module. [typedoc-plugin-missing-exports](https://www.npmjs.com/package/typedoc-plugin-missing-exports) will force typedoc to document that interface even if it isn't exported, but it's better to include it in your entrypoint if possible.

**Comment by kungfooman** [comment]
Hey @Gerrit0, thank you very much! I tested and your suggested plugin works like a charm!

> typedefs will be converted as interfaces -- so long as they are exported.

I'm pretty much new to typedoc, but not new to TypeScript and we discussed this issue back and forth: https://github.com/microsoft/TypeScript/issues/46011

For the time being TypeScript is automatically exporting every `@typedef`, shouldn't we consider this as swapped order incompatibility between `TypeDoc` and `TypeScript`?
